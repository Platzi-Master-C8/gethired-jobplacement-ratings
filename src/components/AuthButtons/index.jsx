import React from 'react';

import Button from '@mui/material/Button';

import { goToEnterprise } from '../../utils';

export const AuthenticationButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Log in
        </Button>
    );
};

export const SignupButton = () => {
    return (
        <Button variant="contained" type="button" onClick={() => goToEnterprise()}>
            Sign up
        </Button>
    );
};
