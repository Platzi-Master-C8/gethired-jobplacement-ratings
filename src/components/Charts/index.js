import RadarChart from "./RadarChart";
import LineChart from "./LineChart";
export { RadarChart, LineChart }