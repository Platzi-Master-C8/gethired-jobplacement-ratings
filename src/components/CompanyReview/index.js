import CompanyReview from './CompanyReview';

export { CompanyReview };
