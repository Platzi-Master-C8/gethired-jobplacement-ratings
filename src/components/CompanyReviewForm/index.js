import CompanyReviewForm from './CompanyReviewForm';

export { CompanyReviewForm };
