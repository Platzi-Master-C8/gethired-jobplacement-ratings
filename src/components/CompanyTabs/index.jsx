import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Tabs, Tab, Box } from '@mui/material';

const TabPanel = ({ children, value, index }) => {
    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`simple-tabpanel-${index}`}
            aria-labelledby={`simple-tab-${index}`}
        >
            {value === index && <Box sx={{ p: 3 }}>{children}</Box>}
        </div>
    );
};

TabPanel.propTypes = {
    children: PropTypes.node.isRequired,
    index: PropTypes.number.isRequired,
    value: PropTypes.number.isRequired,
};

const CompanyTabs = ({ tabsOptions }) => {
    const [value, setValue] = useState(1);

    const handleChange = (_event, newValue) => {
        setValue(newValue);
    };

    return (
        <Box sx={{ width: '100%' }}>
            <Box>
                <Tabs value={value} onChange={handleChange}>
                    {tabsOptions.map((option) => (
                        <Tab key={`TabButton-${option.tabKey}`} label={option.tabKey} />
                    ))}
                </Tabs>
            </Box>
            {tabsOptions.map((option, i) => (
                <TabPanel key={`TabContent-${option.tabKey}`} value={value} index={i}>
                    {option.tabContent}
                </TabPanel>
            ))}
        </Box>
    );
};

CompanyTabs.propTypes = {
    tabsOptions: PropTypes.arrayOf(
        PropTypes.shape({
            tabKey: PropTypes.string.isRequired,
            tabContent: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
        }),
    ).isRequired,
};

export default CompanyTabs;
