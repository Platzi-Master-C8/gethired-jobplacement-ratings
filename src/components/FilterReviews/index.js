import FilterReviews from './FilterReviews';

export {
    FilterReviews
}