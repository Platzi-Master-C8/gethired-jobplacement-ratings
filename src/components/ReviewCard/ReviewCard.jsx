import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Box, Button, Grid, Rating, Typography, Card, CardHeader, CardContent, CardActions } from '@mui/material';
import ThumbUpAltIcon from '@mui/icons-material/ThumbUpAlt';
import ThumbDownAltIcon from '@mui/icons-material/ThumbDownAlt';
import FlagIcon from '@mui/icons-material/Flag';

const StyledRating = styled(Rating)`
    font-size: 1rem;
`;

const HeaderReview = ({ rating }) => (
    <Box>
        <Typography sx={{ ml: 3 }} variant="subtitle1">
            {rating.toFixed(1)}
        </Typography>
        <StyledRating name="read-only" value={rating} readOnly />
    </Box>
);

const SubheaderReview = ({ createdAt, isStillWorkingHere, jobTitle }) => (
    <Box>
        <Typography variant="body2">{jobTitle}</Typography>
        <Typography variant="body2">
            {`(${isStillWorkingHere ? 'Current Employee' : 'Former Employee'})`} - {createdAt}
        </Typography>
    </Box>
);

const ActionsReview = ({ nonUtilityCounter, utilityCounter }) => (
    <CardContent>
        <Typography variant="body1">Was this review helpful?</Typography>
        <CardActions disableSpacing>
            <Button>
                <ThumbUpAltIcon />
                <Typography variant="button2">Yes {utilityCounter}</Typography>
            </Button>
            <Button>
                <ThumbDownAltIcon />
                <Typography variant="button2">No {nonUtilityCounter}</Typography>
            </Button>
            <Button sx={{ ml: 'auto' }}>
                <FlagIcon />
                <Typography variant="button2">Report</Typography>
            </Button>
        </CardActions>
    </CardContent>
);

const ReviewCard = ({ review }) => {
    const {
        contentType,
        createdAt,
        isStillWorkingHere,
        jobTitle,
        nonUtilityCounter,
        utilityCounter,
        weightedAveragePerEvaluation,
    } = review;

    return (
        <Grid item xs={6}>
            <Card sx={{ border: 1, minHeight: 350, maxHeight: 350 }}>
                <CardHeader
                    title={<HeaderReview rating={weightedAveragePerEvaluation} />}
                    subheader={
                        <SubheaderReview
                            createdAt={createdAt}
                            isStillWorkingHere={isStillWorkingHere}
                            jobTitle={jobTitle}
                        />
                    }
                />
                <CardContent sx={{ minHeight: 100, maxHeight: 100 }}>
                    <Typography variant="body1">{contentType}</Typography>
                </CardContent>
                <ActionsReview nonUtilityCounter={nonUtilityCounter} utilityCounter={utilityCounter} />
            </Card>
        </Grid>
    );
};

HeaderReview.propTypes = {
    rating: PropTypes.number.isRequired,
};

SubheaderReview.propTypes = {
    createdAt: PropTypes.string.isRequired,
    jobTitle: PropTypes.string.isRequired,
    isStillWorkingHere: PropTypes.string.isRequired,
};

ActionsReview.propTypes = {
    nonUtilityCounter: PropTypes.number.isRequired,
    utilityCounter: PropTypes.number.isRequired,
};

ReviewCard.propTypes = {
    review: PropTypes.shape({
        contentType: PropTypes.string.isRequired,
        createdAt: PropTypes.string.isRequired,
        isStillWorkingHere: PropTypes.oneOf([0, 1]).isRequired,
        jobTitle: PropTypes.string.isRequired,
        nonUtilityCounter: PropTypes.number.isRequired,
        utilityCounter: PropTypes.number.isRequired,
        weightedAveragePerEvaluation: PropTypes.number.isRequired,
    }).isRequired,
};

export default ReviewCard;
