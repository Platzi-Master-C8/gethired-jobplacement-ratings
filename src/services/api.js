import config from '../config';
import list from '../data/listReviews';
import overallReviews from '../data/overallReviews';

const getData = (endpoint, options) => {
  const ops = {
    headers: config.headers,
    ...options
  };

  const url = `${config.api}/${endpoint}`;
  return fetch(url, ops)
};

const sendData = (endpoint, data) => {
  const url = `${config.api}/${endpoint}`;

  return fetch(url, {
    method: 'POST',
    body: JSON.stringify(data),
    headers: config.headers
  });
}

const api = {
  companyEvaluations: {
    listReviews(companyId, options = {}) {
      // return getData(`companyEvaluations/${companyId}`, options);
      return getData(`companyEvaluations`, options);
    },
    sendReview(companyId, data = {}) {
      return sendData(`companyEvaluations`, data);
    },
    mockDataList() {
      return list.map((review) => ({
        id: review.id,
        contentType: review.content_type,
        createdAt: review.created_at,
        isStillWorkingHere: review.is_still_working_here,
        jobTitle: review.job_title,
        nonUtilityCounter: review.non_utility_counter,
        utilityCounter: review.utility_counter,
        weightedAveragePerEvaluation: review.weighted_average_per_evaluation,
      }));
    },
    mockDataOverallReview() {
      return new Promise(resolve => resolve(...overallReviews));
    }
  },
};

export default api;
