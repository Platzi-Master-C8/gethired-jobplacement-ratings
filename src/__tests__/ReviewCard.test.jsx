import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ReviewCard } from '../components/ReviewCard';
import api from '../services/api';

const defaultProps = {
    reasons: [{ name: 'Suspicious, spam or fake', id: 1 }],
    review: api.companyEvaluations.mockDataList()[0],
};

const renderComponent = (props = {}) => render(<ReviewCard {...defaultProps} {...props} />);

describe('<ReviewCard />', () => {
    test('Component render', () => {
        const { asFragment } = renderComponent({});
        expect(asFragment()).toMatchSnapshot();
    });
});
