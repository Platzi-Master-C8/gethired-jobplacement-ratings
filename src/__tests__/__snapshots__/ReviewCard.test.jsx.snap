// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewCard /> Component render 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xeabas-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 css-wbtyza-MuiTypography-root"
            >
              NA
            </h6>
            <span
              aria-label="0 Stars"
              class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly sc-bdvvtL iXmFMk css-ryrseu-MuiRating-root"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                    data-testid="StarBorderIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            />
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              (Former Employee) - Feb 04, 2022
            </p>
          </div>
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-1e29m5v-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      />
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Was this review helpful?
      </p>
      <div
        class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
      >
        <div
          class="MuiToggleButtonGroup-root css-1ros84a-MuiToggleButtonGroup-root"
          role="group"
        >
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-primary MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-d9c359-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="like"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ThumbUpAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2 20h2c.55 0 1-.45 1-1v-9c0-.55-.45-1-1-1H2v11zm19.83-7.12c.11-.25.17-.52.17-.8V11c0-1.1-.9-2-2-2h-5.5l.92-4.65c.05-.22.02-.46-.08-.66-.23-.45-.52-.86-.88-1.22L14 2 7.59 8.41C7.21 8.79 7 9.3 7 9.83v7.84C7 18.95 8.05 20 9.34 20h8.11c.7 0 1.36-.37 1.72-.97l2.66-6.15z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-button2 css-37jdci-MuiTypography-root"
            >
              Yes
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-pressed="false"
            class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium Mui-error MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-17pma60-MuiButtonBase-root-MuiToggleButton-root"
            tabindex="0"
            type="button"
            value="dislike"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ThumbDownAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 4h-2c-.55 0-1 .45-1 1v9c0 .55.45 1 1 1h2V4zM2.17 11.12c-.11.25-.17.52-.17.8V13c0 1.1.9 2 2 2h5.5l-.92 4.65c-.05.22-.02.46.08.66.23.45.52.86.88 1.22L10 22l6.41-6.41c.38-.38.59-.89.59-1.42V6.34C17 5.05 15.95 4 14.66 4h-8.1c-.71 0-1.36.37-1.72.97l-2.67 6.15z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-button2 css-37jdci-MuiTypography-root"
            >
              No
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textInfo MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-136jskx-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FlagIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.4 6 14 4H5v17h2v-7h5.6l.4 2h7V6z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-button2 css-37jdci-MuiTypography-root"
          >
            Report
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
